projects/baby_driver/scripts/gpio_interrupts.py

```diff
@@ -0,0 +1,182 @@
+"""Python implementation of register_gpio_interrupt function"""
+import can_util
+from gpio_port import GpioPort
+import message_defs
+
+NUM_PINS_PER_PORT = 16
+
+# pylint: disable=too-few-public-methods
+class InterruptEdge:
+    """Stores interrupt edge values"""
+    RISING = 0
+    FALLING = 1
+    RISING_AND_FALLING = 2
+    NUM_INTERRUPT_EDGES = 3
+
+# Callbacks that user defines when registering interrupt are stored here
+callback_dict = {
+    #("port", "pin"): callback,
+}
+
+def callback_listener(can_message):
+    """
+    Called by the notifier everytime a message is received from the firmware side notifiying
+    of an interrupt. It filters through the messages to find the can message with babydriver
+    message ID GPIO_IT_INTERRUPT and calls user-defined/default callback that is registered
+    with the interrupt
+    """
+
+    unfiltered_msg = can_util.Message.from_msg(can_message)
+    # print(f"{unfiltered_msg.message_id}, {unfiltered_msg.data[0]},
+    # {unfiltered_msg.data[1]}, {unfiltered_msg.data[1]}")
+    if (unfiltered_msg.message_id == message_defs.BABYDRIVER_CAN_MESSAGE_ID and
+        unfiltered_msg.data[0] == message_defs.BabydriverMessageId.GPIO_IT_INTERRUPT):
+
+        data = unfiltered_msg.data[:]
+        port = data[1]
+        pin = data[2]
+        edge = data[3]
+        # Calling function from callback_dict based on (port, pin)
+        try:
+            # ret is only used for testing purposes
+            ret = callback_dict[(port, pin)]((port, pin, edge))
+        except TypeError as type_error:
+            raise "Callback function parameters are of incorrect format (use a named-tuple called \
+                    info to store port, pin and edge" from type_error
```

Comment:
You can't raise strings in Python (that's only JS afaik), so you'll need to wrap this in an exception type like `ValueError` or `TypeError`. (The `raise ... from` syntax only sets the cause of the new error to the first error.)

Also when breaking Python strings over multiple lines, it's good practice to do it like this:
```py
a = ("some really really long string"
     " which is really really long")
```
Your current method with a backslash escape actually includes all the indentation in the second line in the string, so the string actually has a ton of whitespace in the middle.